spec/integration/aws_rds_spec.rb

```diff
@@ -0,0 +1,97 @@
+require 'spec_helper'
+require 'cheffish/rspec/chef_run_support'
+require 'chef/provisioning/aws_driver/credentials'
+
+describe 'Aws RDS DB instance' do
+  extend Cheffish::RSpec::ChefRunSupport
+
+  when_the_chef_12_server 'exists' do
+    organization 'foo'
+
+    let(:rds_client) { double(AWS::RDS::Client) }
+    let!(:entry_store) { Chef::Provisioning::ChefManagedEntryStore.new }
+
+    before :each do
+      Chef::Config.chef_server_url = URI.join(Chef::Config.chef_server_url, '/organizations/foo').to_s
+      allow_any_instance_of(AWS.config.class).to receive(:rds_client).and_return(rds_client)
+      allow(Chef::Provisioning::ChefManagedEntryStore).to receive(:new).and_return(entry_store)
+      allow_any_instance_of(Chef::Provisioning::AWSDriver::Credentials).to receive(:default)
+        .and_return({
+          :aws_access_key_id => 'na',
+          :aws_secret_access_key => 'na'
+        })
+    end
+
+    describe 'action :create' do
+
+      before do
+        resp = AWS::Core::Response.new
+
+        expect(rds_client).to receive(:describe_db_instances)
+          .with({:db_instance_identifier=>'my-db-instance'})
+          .and_raise(AWS::RDS::Errors::DBInstanceNotFound)
+
```

Comment:
Just noting: expectations in before blocks can work just fine; it can be an easy way to make sure the before block set up the test appropriately, even though the setup isn't the test itself.